tests/components/pooldose/test_sensor.py

```diff
@@ -156,39 +144,128 @@ async def test_sensor_entity_unavailable_no_coordinator_data(
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Verify sensor uses unit from fixture data
+    flow_state = hass.states.get("sensor.pool_device_flow_rate")
+    assert flow_state is not None
+    assert flow_state.attributes["unit_of_measurement"] == "L/s"
+    assert flow_state.state == "150"
+
 
-    # Change to Fahrenheit via mock update
-    instant_values_raw = await async_load_fixture(hass, "instantvalues.json", DOMAIN)
-    updated_data = json.loads(instant_values_raw)
-    updated_data["sensor"]["temperature"]["unit"] = "°F"
-    updated_data["sensor"]["temperature"]["value"] = 77
+@pytest.mark.usefixtures("mock_pooldose_client")
+async def test_sensor_unit_fallback_to_description(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_pooldose_client: AsyncMock,
+) -> None:
+    """Test sensor falls back to entity description unit when no dynamic unit."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
 
+    # Test ORP sensor - should use static unit from entity description
+    orp_state = hass.states.get("sensor.pool_device_orp")
+    assert orp_state is not None
+    assert orp_state.attributes["unit_of_measurement"] == "mV"
+    assert orp_state.state == "718"
+
+    # Test pH sensor - should have no unit (None in description)
+    ph_state = hass.states.get("sensor.pool_device_ph")
+    assert ph_state is not None
+    assert "unit_of_measurement" not in ph_state.attributes
+    assert ph_state.state == "6.8"
+
+
+@pytest.mark.usefixtures("mock_pooldose_client")
+async def test_sensor_no_data_returns_none_unit(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_pooldose_client: AsyncMock,
+) -> None:
+    """Test sensor returns None unit when no data available."""
+    # Mock API response with missing temperature and flow_rate data
     mock_pooldose_client.instant_values_structured.return_value = (
         RequestStatus.SUCCESS,
-        updated_data,
+        {"sensor": {}, "binary_sensor": {}, "number": {}, "switch": {}, "select": {}},
     )
 
-    freezer.tick(timedelta(minutes=10))
-    async_fire_time_changed(hass)
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
     await hass.async_block_till_done()
 
-    # Check unit changed to Fahrenheit
+    # Verify no temperature entities are created when no data
     temp_state = hass.states.get("sensor.pool_device_temperature")
-    # After reload, the original fixture data is restored, so we expect °C
-    assert temp_state.attributes["unit_of_measurement"] == UnitOfTemperature.CELSIUS
-    assert temp_state.state == "25.0"  # Original fixture value
+    # Entity may be created but should be unavailable or have no unit
+    if temp_state:
+        assert temp_state.state == "unavailable"
```

Comment:
we shouldn't have if statements in the tests

Reply:
no longer in code